### PR TITLE
BUG: Throw error on invalid path in save_figure

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to accomplish 2 things:

* Throw explicit exception when a given path to save_figure in Export Plot plugin contains directory that does not exist (instead of failing silently).
* Also refactor the plugin to use `pathlib.Path` as @duytnguyendtn wanted.

I added tests for the exceptions but I cannot test the actual file saving functionality because CI will hang when the saving is farmed out to JS by bqplot (https://github.com/bqplot/bqplot/pull/1397/files#r726500097).

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-3655)

### CODE REVIEW

File: jdaviz/configs/default/plugins/export_plot/export_plot.py [hunk 1]
```diff
@@ -89,26 +90,37 @@ def save_figure(self, filename=None, filetype=None):
...
             if filename is None:
                 viewer.figure.save_png()
             else:
+                if not filename.parent.exists():
+                    raise ValueError(f"Invalid path={filename.parent}")
+
                 # support writing without save dialog
                 # https://github.com/bqplot/bqplot/pull/1397
                 def on_img_received(data):
-                    with open(os.path.expanduser(filename), 'bw') as f:
```
Comment (pllim) [inline jdaviz/configs/default/plugins/export_plot/export_plot.py:107]: @kecnry , I don't remember why you added `expanduser` here. What was the use case that triggered this? Using Path, I am hoping we don't need it anymore but should make sure.

Comment (pllim) [reply jdaviz/configs/default/plugins/export_plot/export_plot.py:107]: Nvm... Found the original issue https://github.com/spacetelescope/jdaviz/pull/929#issuecomment-1550331365

**Comment by pllim** [comment]
devdeps failure unrelated, codecov complain unavoidable

**Review by rosteen — APPROVED** [review]
I'm not sure that I hit every way there is to trigger this, but it seems to work.

**Review by kecnry — APPROVED** [review]
LGTM, thanks for the quick fix!

**Pull request merged**